src/Reflectivity-VariableBreakpoint-Extensions/ClyBreakOnVarAccessCommand.class.st

```diff
@@ -0,0 +1,25 @@
+"
+I break on all accesses to selected target variable in the source code
+"
+Class {
+	#name : #ClyBreakOnVarAccessCommand,
+	#superclass : #ClyVariableBreakpointCommand,
+	#category : #'Reflectivity-VariableBreakpoint-Extensions'
+}
+
+{ #category : #testing }
+ClyBreakOnVarAccessCommand class >> canBeExecutedInContext: aSourceCodeContext [
+	^ (super canBeExecutedInContext: aSourceCodeContext)
+		and: [ aSourceCodeContext selectedSourceNode isArgument not ]
```

Comment:
Why it does not work for arguments?

Reply:
It is because of how I define "access", for me it is write + read, and as we cannot write in args, then only the read option is available. 
It could be changed, by I feel it is not worth the effort.

Reply:
I see. I think it deserves the comment.

From the other side it can be confusing to the users of UI:
For the most cases there will be a "break on access". Then when occasionally I will select a variable (without thinking that it is an argument) that would be quite unexpected to not find this command. 
I would really keep this command for all variables even if "break on read" would do the same.

Reply:
But then users will be confused because they won't understand the difference between "access" and "read". I feel we can always make an unperfect choice, and users will have to learn.
But the point is good, so what could be done:
- add in the name of the menu item "arg" or "inst" or "temp" to be explicit
- mention the subtility in the popover help

Reply:
> But then users will be confused because they won't understand the difference between "access" and "read".

Personally I do not see the confusion here. Users need to understand the difference between these two commands independently from the argument type.
The fact that argument variables are not writable anywhere in the method should not break this understanding IMHO. Meaning of "access" word here is not changed (from the user point of view).

Reply:
Also arguments can be writable. There is a settings in the compiler. it can be useful for code migration from old versions of Pharo or Squeak

Reply:
Thanks, I'll give you a reproducible case. It is possible I am using it the wrong way.

Reply:
For now I do not wish to spend more time on this, I think we should do an issue on this and brainstorm from there.
My choices can be discussed w/o any problem, I don't claim I make the best ones ;) but I will not be able to put more efforts right now and it will block the PR.